ref(apple): Mention edge cases when OOM tracking is not working (#4167)
getsentry/sentry-docs: src/platforms/apple/common/configuration/out-of-memory.mdx

This integration tracks out-of-memory (OOM) crashes based on heuristics. This feature is available for iOS, tvOS, and Mac Catalyst and works only if the application was in the foreground.

When your application uses too much RAM, the operating system kills your app without a normal termination process. Your process doesn't receive a signal or any other type of information to ensure an OOM occurs. As a result, in the Apple SDK, we track out-of-memory crashes during the app start based on heuristics.

> The SDK checks the following statements when the user launches the app and reports an OOM if all of them are true:

Reviewer: ```suggestion
When a user launches the app, the SDK checks the following statements and reports an OOM if all of them are true:
```

* The app didn't crash on the previous run.

> * The app was in the foreground / active.

Reviewer: ```suggestion
* The app was in the foreground/active.
```

* The user launched the app at least once. When your app produces an OOM during the first launch, the SDK can't detect it.

> * There was no debugger attached to the process of the app. Our logic would falsely report OOMs whenever you restart the app in development.

Reviewer: I don't really understand how the second sentence fits. Have made a suggestion for clarity.

```suggestion
* There was no debugger attached to the process of the app. If there was, our logic would falsely report OOMs whenever you restarted the app in development.
```

* The OS did not update your app.

> * The user didn't update the OS of his device.

Reviewer: inclusive language

```suggestion
* The user didn't update the OS of their device.
```
